imports/plugins/core/accounts/client/containers/addressBookContainer.js

```diff
@@ -0,0 +1,214 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { compose } from "recompose";
+import { Meteor } from "meteor/meteor";
+import { Template } from "meteor/templating";
+import { i18next } from "/client/api";
+import { Countries } from "/client/collections";
+import * as Collections from "/lib/collections";
+import { registerComponent, composeWithTracker } from "@reactioncommerce/reaction-components";
+import AddressBook from "../components/addressBook";
+
+/**
+ * @method updateAddress
+ * @summary helper function that updates an address in the account's addressBook via a meteor method.
+ * @since 2.0.0
+ * @param {Object} address - address to be updated.
+ * @param{String} property - property to be updated.
+ * @return {Promise}
+ */
+function updateAddress(address, property) {
+  return new Promise((resolve, reject) => {
+    if (property) {
+      Meteor.call("accounts/addressBookUpdate", address, null, property, (error, result) => {
+        if (error || !result) reject(i18next.t("addressBookGrid.somethingWentWrong", { err: error.message }));
+        if (result) resolve(result);
+      });
+    } else {
+      Meteor.call("accounts/addressBookUpdate", address, (error, result) => {
+        if (error || !result) reject(i18next.t("addressBookGrid.somethingWentWrong", { err: error.message }));
+        if (result) resolve(result);
+      });
+    }
+  });
+}
+
+/**
+ * @method removeAddress
+ * @summary helper function that updates an address in the account's addressBook via a meteor method.
+ * @since 2.0.0
+ * @param {String} _id - _id of address to be removed.
+ * @return {Promise}
+ */
+function removeAddress(_id) {
+  return new Promise((resolve, reject) => {
+    Meteor.call("accounts/addressBookRemove", _id, (error, result) => {
+      if (error || !result) reject(i18next.t("addressBookGrid.cantRemoveThisAddress", { err: error.message }));
+      if (result) resolve(result);
+    });
+  });
+}
+
+/**
+ * @method addAddress
+ * @summary helper function that adds an address in the account's addressBook via a meteor method.
+ * @since 2.0.0
+ * @param {Object} address - address to be added.
+ * @return {Promise}
+ */
+function addAddress(address) {
+  return new Promise((resolve, reject) => {
+    Meteor.call("accounts/validateAddress", address, (error, result) => {
+      if (error || !result) reject(i18next.t("addressBookAdd.failedToAddAddress", { err: error.message }));
+      if (result && result.validated) {
```

Comment:
This second `if` leaves the possibility of a swallowed error, when `result.validated` is falsy. I think it'd be better to remove this second `if` and change above to `if (error || !result || !result.validated)`